pyiron/base/generic/hdfio.py

```diff
@@ -1252,14 +1252,51 @@ def create_object(self, class_name, **qwargs):
         return getattr(
             importlib.import_module(module_path),
             internal_class_name,
-        )(**qwargs)
+        )
+
+    @staticmethod
+    def make_from_hdf(cls, **kwargs):
+        """
+        Create new instance of the given class from HDF5 file.
+
+        Uses the given **kwargs and a special classmethod "from_hdf_args" that
+        may be defined on cls to construct a dictionary of arguments and then
+        instatiate cls with them.
+
+        kwargs must contain all necessary arguments for `cls.__init__`, even
+        positional ones.  They will be passed by name, i.e. `cls.__init__` must
+        not have positional-only parameters to be correctly instantiated by
+        this method.
+
+        Args:
+            cls (type): pyiron type to instantiate
+            **kwargs: arguments for instance creation
+        """
+
+        # kwargs might provide more arguments than needed for this class, since
+        # it has to provide init arguments for all possible pyiron types.  Here
+        # we just pick the ones needed for this specific instance based on the
+        # signature for cls.__init__
+        if hasattr(cls, "from_hdf_args"):
+            init_args = cls.from_hdf_args(hdf)
```

Comment:
Currently only the job object has default arguments, while all other objects can be initialize with empty arguments and the call `from_hdf()`. I would prefer to initialize all objects with empty arguments and then load everything in the `from_hdf()`.

Reply:
Fair, does that work for jobs or is it easy to make it work?  This is just me trying to be future-proof, I didn't have a use case in mind yet.

Reply:
jobs get the `job_name` and `project_hdf5` as input - which could be loaded from the HDF5 file as well.